### PR TITLE
Set a minimum size for status panel elements in ScriptCreateDialog

### LINKED ISSUE #48481: "Attach Node Script" popup is stretched across entire viewport height
**Godot version:**
3.4.beta 1209ee0df

**OS/device including version:**
Windows 10.

**Issue description:**

image

**Steps to reproduce:**
1. Create new scene with a root node.
2. Try to attach a new script to root node.

**Minimal reproduction project:**
Not needed.

### PR DESCRIPTION
**Pull request by aaronfranke**
Hopefully fixes #48481 for real this time.

### CODE REVIEW

File: editor/script_create_dialog.cpp [hunk 1]
```diff
@@ -758,6 +759,7 @@ ScriptCreateDialog::ScriptCreateDialog() {
 	/* Error Messages Field */
 
 	VBoxContainer *vb = memnew(VBoxContainer);
+	vb->set_custom_minimum_size(Vector2(340, 30));
```
Comment (pycbouh) [inline editor/script_create_dialog.cpp:762]:
```suggestion
	vb->set_custom_minimum_size(Size2(340, 30) * EDSCALE);
```

File: editor/script_create_dialog.cpp [hunk 2]
```diff
@@ -766,13 +768,15 @@ ScriptCreateDialog::ScriptCreateDialog() {
 	vb->add_child(path_error_label);
 
 	builtin_warning_label = memnew(Label);
+	builtin_warning_label->set_custom_minimum_size(Vector2(340, 10));
```
Comment (pycbouh) [inline editor/script_create_dialog.cpp:771]:
```suggestion
	builtin_warning_label->set_custom_minimum_size(Size2(340, 10) * EDSCALE);
```

File: editor/script_create_dialog.cpp [hunk 3]
```diff
@@ -766,13 +768,15 @@ ScriptCreateDialog::ScriptCreateDialog() {
...
 	builtin_warning_label->set_text(
 			TTR("Note: Built-in scripts have some limitations and can't be edited using an external editor."));
 	vb->add_child(builtin_warning_label);
 	builtin_warning_label->set_autowrap(true);
 	builtin_warning_label->hide();
 
 	script_name_warning_label = memnew(Label);
+	script_name_warning_label->set_custom_minimum_size(Vector2(340, 10));
```
Comment (pycbouh) [inline editor/script_create_dialog.cpp:779]:
```suggestion
	script_name_warning_label->set_custom_minimum_size(Size2(340, 10) * EDSCALE);
```

File: editor/script_create_dialog.cpp [hunk 4]
```diff
@@ -781,6 +785,7 @@ ScriptCreateDialog::ScriptCreateDialog() {
 	script_name_warning_label->hide();
 
 	status_panel = memnew(PanelContainer);
+	status_panel->set_custom_minimum_size(Vector2(350, 40));
```
Comment (pycbouh) [inline editor/script_create_dialog.cpp:788]:
```suggestion
	status_panel->set_custom_minimum_size(Size2(350, 40) * EDSCALE);
```

**Review by YuriSizov — CHANGES_REQUESTED** [review]
You need to use `Size2` (for consistency at least), and multiply by `EDSCALE`.

However, I'm not convinced it's a proper solution, even if it works at the first glance. It seems that hidden autowrapped labels do indeed tend to be one character wide and therefore incredibly high. But the layout of this dialog is just a mess, IMO. A lot of unnecessary nodes.

**Review by YuriSizov — APPROVED** [review]

**Comment by pycbouh** [comment]
That said, it does seem to address the problem (and a similar one in the FileSystem dock). But something is very wrong if we need to set minimal sizes on labels that are supposed to be expanded by their parent in order to avoid such bugs.

Also it may make sense to use a single instance of `ScriptCreateDialog` in the editor (we currently have 3 in 3 different places).

**Comment by pycbouh** [comment]
Agreed. By the way, are we sure this problem doesn't manifest in master?

**Comment by aaronfranke** [comment]
@pycbouh It doesn't. Why it happens in 3.x and not master, I have no idea. It could be argued that it's worth making this change in master as a precaution just in case there is some edge case where it does happen.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**